core/templates/services/favicon.service.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for changing favicon dynamically.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FaviconService {
+  constructor() {}
+
+  setFavicon(faviconUrl: string): void {
+    const link: HTMLLinkElement = (
+      document.querySelector('link[rel*="icon"]') ||
+      document.createElement('link'));
+    link.href = faviconUrl;
+    link.type = 'image/x-icon';
+    link.rel = 'icon';
+    document.getElementsByTagName('head')[0].appendChild(link);
```

Comment:
This looks a bit odd -- I don't see any deletion of the previous element that was already there, just appending. What happens to the previous element, shouldn't it be removed?

Reply:
I have added the removal part now. But, favicon was changing even before removing.